src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,110 @@
-export const Pagination = () => {};
+import cn from 'classnames';
+import { getNumbers } from '../../utils';
+
+interface Props {
+  total: number,
+  perPage: number,
+  currentItem: number,
+  onPageChange: (page: number) => void,
+}
+
+export const Pagination: React.FC<Props> = (props) => {
+  const {
+    total,
+    perPage,
+    currentItem,
+    onPageChange,
+  } = props;
+
+  const maxItemOnCurPage = currentItem + perPage - 1 > total
+    ? total : currentItem + perPage - 1;
+
+  const pageNumberHandler = (n: number) => {
+    if (currentItem === 1 + perPage * (n - 1)) {
+      return;
+    }
+
+    onPageChange(1 + perPage * (n - 1));
+  };
```

Comment:
```suggestion
  const handlePageNumberChange = (n: number) => { // I don't know what is `n` in your case, rename it
    const someCalculatedVariable = 1 + perPage * (n - 1). // The same with this variable
    if (currentItem !== someCalculatedVariable ) {
      onPageChange(someCalculatedVariable);
    }
  };
```
- Change handlers' names in all places.
- Avoid one letter variables such as `n`.
- You do calculations two times `1 + perPage * (n - 1)`. Create a variable.
- Also, you can do some action only if condition is true and it will be less code

**Fix issues in all places**